src/Compilers/CSharp/Portable/FlowAnalysis/ControlFlowPass.cs

```diff
@@ -232,7 +232,11 @@ protected override void VisitStatement(BoundStatement statement)
 
         private void CheckReachable(BoundStatement statement)
         {
-            if (!this.State.Alive && !this.State.Reported && !statement.WasCompilerGenerated && statement.Syntax.Span.Length != 0)
+            if (!this.State.Alive &&
+                !this.State.Reported &&
+                !statement.WasCompilerGenerated &&
+                statement.Kind != BoundKind.LocalFunctionStatement &&
+                statement.Syntax.Span.Length != 0)
```

Comment:
You can accomplish this effect by adding a `case` in the previous method (that is sort of the purpose of the previous method).